### DISCUSSION TITLE
Seeing other people's file paths in the suggestions

### DISCUSSION THREAD

**Discussion by OskarLJ** [community/community#63722] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I'm getting suggestions for file import with other people's file paths and it just feels wrong and like a privacy issue. Mostly, this is not exactly private data and I guess it's not extraordinary sensitive. I could still imagine that its a problem, e.g. if your folder names contain project names that are not supposed to be public. I'd assume that the folder name was already public somehow to have CoPilot suggest it, but still.
Despite being absolutely useless and wrong, it also makes one worry for a bit if someone else will get suggestions with my file paths in the future (From what I got this should not be possible?).
Here's an example for a suggestion that I just got:

> `# set the path to the folder containing the data`
> `path = r'C:\Users\karen\OneDrive - University of Glasgow\PhD\PhD_Progect\Peak_analysis_final\Areas'`

I think it's weird that I didn't find anything about it in the discussions. Am I the only one seeing this stuff or am I the only one considering it problematic? I'm new to CoPilot, I'm still unsure what to expect and how to properly utilize it.

**Comment by Akash1134** [comment]
Hi @OskarLJ ,

I understand your concern about GitHub Copilot suggesting file paths that may contain sensitive information. While GitHub Copilot is designed to suggest code based on patterns and structures in the code you've already written, it's possible that it may suggest file paths that contain sensitive information.

It's important to note that GitHub Copilot is an AI-powered tool that learns from the code that it's trained on, and it doesn't have access to any private or sensitive information. However, it's still possible that it may suggest file paths that contain sensitive information if those file paths are included in the code that it's trained on.

If you're concerned about the file paths that GitHub Copilot is suggesting, you can try the following steps:

1. Review the code that you've written: Review the code that you've written to see if it contains any sensitive information, such as file paths or other private data. If you find any sensitive information, you can remove it or replace it with dummy data.

2. Use dummy data: If you're working with sensitive data, you can use dummy data or placeholders in your code to avoid exposing sensitive information.

3. Be mindful of what you share: Be mindful of what you share in your code and avoid including sensitive information, such as file paths or other private data.

4. Report any issues: If you encounter any issues with GitHub Copilot suggesting sensitive information, you can report them to GitHub support. They may be able to provide more specific guidance on how to resolve the issue based on your particular setup and configuration.

Overall, while GitHub Copilot is designed to suggest code based on patterns and structures in the code you've already written, it's possible that it may suggest file paths that contain sensitive information. By reviewing your code, using dummy data, being mindful of what you share, and reporting any issues, you can help protect your sensitive information and ensure that GitHub Copilot is working properly in your workflow.

**Reply by Rockyread** [reply]
Fake Akash

**Reply by Akash1134** [reply]
Hi @Rockyread ,

It is an illustration of a bad example of using AI tools to respond to any questions/feedback in the community. Please see my response below. 🙂

**Comment by Akash1134** [comment]
Hi @OskarLJ ,

Please note that the above response is generated using an AI tool to showcase example of bad usage of AI generated content. The response may be used as an example for other community members that using ChatGPT or other AI generated tools without researching  and adding real value can provide inaccurate information for members of this community. Some of the risks and limitations that such AI generated tools might pose are:

- "May occasionally generate incorrect information"
- "May occasionally produce harmful instructions"
- "Limited knowledge of world and events after 2021"


Please  ignore the above response and refer the below response to your issue.

Thanks!

**Comment by Akash1134** [comment]
Hi @OskarLJ ,

### Response to your question above

Because Codex, the model powering GitHub Copilot, was trained on publicly available code, its training set included personal data that was included in that code. From our internal testing, we found it to be very rare that GitHub Copilot suggestions included personal data verbatim from the training set.

In some cases, the model will suggest what appears to be personal data – email addresses, phone numbers, etc. – but those suggestions are actually fictitious information synthesized from patterns in training data and therefore do not relate to any particular individual. For example, when one of our engineers prompted GitHub Copilot with, “My name is Mona and my birthdate is,” GitHub Copilot suggested a random, fictitious date of “December 12,” which is not Mona’s actual birthdate.

We have also implemented a filter that blocks emails when shown in standard formats, but it’s still possible to get the model to suggest this sort of content if you try hard enough. We will keep improving the filter system to be more intelligent to detect and remove more personal data from the GitHub Copilot suggestions.


#### [More on GitHub Copilot’s always-on security filter for Suggestions](https://github.blog/2023-02-14-github-copilot-now-has-a-better-ai-model-and-new-capabilities/)

**Comment by Jaber-Saed** [comment]
I understand your concern about CoPilot suggesting file paths that may contain potentially sensitive or private information. While CoPilot is designed to assist developers by providing code suggestions, it's important to be cautious when sharing code that includes personal or sensitive information, even in file paths.

Here are a few points to consider:

1. **Sensitivity of Information:** While the example you've provided doesn't seem to include highly sensitive information, there might be cases where file paths could inadvertently reveal more private information, such as usernames, project names, or proprietary folder structures.

2. **Contextual Understanding:** CoPilot generates suggestions based on patterns in code it has seen during training. It's possible that some file paths mentioned in suggestions were common in open-source repositories used during training. CoPilot may not fully understand the context or privacy implications of those paths.

3. **Risk Mitigation:**
   - Review Suggestions: Always review code suggestions made by CoPilot before incorporating them into your codebase.
   - Modify Suggestions: If you see sensitive information in the suggestion, consider modifying the suggestion to remove any private data before using it.

4. **Feedback to OpenAI:**
   - If you encounter suggestions that you find problematic, providing feedback to OpenAI could help improve the system's behavior over time. OpenAI has mechanisms to incorporate user feedback for system improvements.

5. **Best Practices:**
   - Avoid sharing code with potentially sensitive information in public spaces like open-source repositories, forums, or code-sharing platforms.

6. **Educational Use:**
   - Keep in mind that CoPilot's suggestions are based on a broad range of training data. Some of these suggestions might not be aligned with your specific project's requirements, especially if your project involves proprietary or confidential data.

Remember that technology, including AI models like CoPilot, is continually evolving. While CoPilot's suggestions can be valuable for productivity, it's important to exercise discretion and be mindful of the information you share in your code and other online interactions.

As you continue using CoPilot, it's a good practice to be vigilant about potential privacy concerns and ensure that the code you integrate or share aligns with your intended level of privacy and security.
